pkgs/development/compilers/solc/default.nix

```diff
@@ -10,69 +20,88 @@ assert z3Support -> z3 != null && lib.versionAtLeast z3.version "4.6.0";
 assert cvc4Support -> cvc4 != null && cln != null && gmp != null;
 
 let
-  jsoncppVersion = "1.9.4";
+  jsoncppVersion = "1.9.3";
   jsoncppUrl = "https://github.com/open-source-parsers/jsoncpp/archive/${jsoncppVersion}.tar.gz";
   jsoncpp = fetchzip {
     url = jsoncppUrl;
-    sha256 = "0qnx5y6c90fphl9mj9d20j2dfgy6s5yr5l0xnzid0vh71zrp6jwv";
+    sha256 = "1vbhi503rgwarf275ajfdb8vpdcbn1f7917wjkf8jghqwb1c24lq";
   };
-in
-gccStdenv.mkDerivation rec {
-
-  pname = "solc";
-  version = "0.7.4";
 
-  # upstream suggests avoid using archive generated by github
-  src = fetchzip {
-    url = "https://github.com/ethereum/solidity/releases/download/v${version}/solidity_${version}.tar.gz";
-    sha256 = "02261l54jdbvxk612z7zsyvmchy1rx4lf27b3f616sd7r56krpkg";
+  range3Version = "0.11.0";
+  range3Url = "https://github.com/ericniebler/range-v3/archive/${range3Version}.tar.gz";
```

Comment:
Should we take `range-v3` as an input and then use its `.src` and `.version` attributes?
The current [range-v3 derivation]((https://github.com/NixOS/nixpkgs/blob/cad078aa872badd36b4b60fa669a0281a664fa99/pkgs/development/libraries/range-v3/default.nix#L11)
) has the same version/sha256 as this fetch

Reply:
since they are pinned independently, I would like to keep them separate.

I already had some issues using a different  version of jsoncpp, I would like to minimize the drift from upstream.